bcdc_apitests/tests/resources/test_resources.py

```diff
@@ -23,87 +23,141 @@ def test_resource_create(ckan_url, ckan_rest_dir, ckan_auth_header,
+
+        # get pkg id
         resource_data['package_id'] = pkg['id']
-        resp = requests.post(api_call, headers=ckan_auth_header, json=resource_data)
-        logger.debug("resource_create: %s", resp.text)
-        assert resp.status_code == 200
-        # TODO: Follow this up with a resource_show
+
+        # create resource
+        res_data = requests.post(api_call, headers=ckan_auth_header, json=resource_data)
+        logger.debug("resource_create: %s", res_data.text)
+
+        # get resource id
+        resp_json = res_data.json()
+        resId = resp_json['result']['id']
+
+        assert res_data.status_code == 200
+
+        # check is resource exist
+        # define remote api
+        api_call = '{0}{1}/{2}'.format(ckan_url, ckan_rest_dir, 'resource_show')
+        logger.debug('api_call: %s', api_call)
+        res_data = requests.post(api_call, headers=ckan_auth_header, json={'id': resId})
+        logger.debug("resource_show: %s", res_data.text)
+
+        resp_json = res_data.json()
+
+        assert resp_json['success']
+
     except ckanapi.CKANAPIError as err:
         logger.debug("err: %s %s", type(err), err)
         raise err
 
 
 # update resource
-def test_resource_update(remote_api_admin_auth, test_package_name, resource_data):
-    remote_api = remote_api_admin_auth
-    logger.debug("resource_name: %s", resource_data['name'])
+def test_resource_update(remote_api_admin_auth,ckan_url,ckan_rest_dir,ckan_auth_header,resource_data):
+    '''
+    :param remote_api_admin_auth: ckanapi remote, with auth
+    :param resource_data: test resource structure
+    :param ckan_url:
+    :param ckan_rest_dir:
+    :param ckan_auth_header:
+    '''
+    try:
 
-    # using search to get the ID of resource.  needs to migrate as fixture via package query
-    # TODO: Doesn't look like this is working anymore, doesn't return any records.
-    res_data = remote_api.action.resource_search(query="name:{0}".format(resource_data['name']))
-    logger.debug("res_data: %s", res_data)
-    
-    #res_id = res_data['results'][0]['id']
-    # res_id = res_data['id']
+        # define remote api
+        remote_api = remote_api_admin_auth
+        logger.debug("resource_name: %s", resource_data['name'])
 
-    #logger.debug("resource_id: %s", res_id)
+        # using search to get the ID of resource.
+        res_data = remote_api.action.resource_search(query="name:{0}".format(resource_data['name']))
 
-    try:
-        res_data = remote_api.action.resource_update(**resource_data)
-        logger.debug("resource_update: %s", res_data)
-        # TODO: KN, put in an assertion here that verifies that you successfully updated
+        # update resource data dic with package id
+        resource_data['package_id'] = res_data['results'][0]['package_id']
+        logger.debug("resource_pkgID: %s", resource_data['package_id'])
+
+        # add id key to resource data dic
+        resource_data['id'] = res_data['results'][0]['id']
```

Comment:
thinking you could assert that the resource was successfully before this line... then in the event that the resource isn't found you won't get an index out of bounds error, but instead an assertion failure.   Same thing on line 86 where you are getting the package id... assert that you have results before you get that value.... just an idea